Gemfile.lock

```diff
@@ -275,7 +276,9 @@ GEM
       rex-socket
       rex-text
     rex-struct2 (0.1.2)
-    rex-text (0.2.17)
+    rex-text (0.2.18)
+      openssl
```

Comment:
`rex-text` now depends on `openssl` and `ruby-rc4` was this really the place this should be added?

Looking at the code this looks to be done because base64 encode was already in rex-text, however it does not seem like a text parsing and manipulation library should be preforming encryption operations that make convert the object from `text` format to a `binary` format.

Reply:
Answer to 1st question: So I added that dependency in rex-text. But then when I was trying bump rex-text's version in Framework's gemfile, it automatically modified that. Let me know what the right way is and then I will update it

Answer to the 2nd question: Rex has a thing called binary_manipulation.rb... so perhaps you can convert text to binary? I'm not trying to be defensive here. If you don't feel the encryption should be in rex-text, please let me know how you'd like to implement so I could make changes. Thanks.

Reply:
Sorry I should be more clear, this dependency was added due to rapid7/rex-text/pull/13 importing openssl and ruby-rc4 classes.  

Unfortunately I did not review that PR.  I am suggesting that we should refactor that change and not introduce cryptographic functionality in the `rex-text` library as part of the `Rex::Text` namespace.

Having reviewed `binary_manipulation.rb` most of those methods in the `Rex::Text` namespace look to have been intended to get `Text` type.  I am not defending that the usage and the namespace may already be overloaded some. I am simply suggesting not to extend this in this way.  As I see it a *Text Manipulation* library would not do encryption tasks.  That should be the responsibility of some other library or tool in another namespace.

Reply:
I agree.

If you want, we can do a `Rex::Encryptor` or something like that.

Reply:
That sounds good, we can add that namespace in the framework repo for now and if it grows we can extract as an external library later.  

I think `xor` and `base64` are reasonable as "text" manipulations to stay in `rex-text` for now if you prefer to leave them.

Reply:
Understood. I'll update the PR. Thanks!